### DISCUSSION TITLE
In Lesson-4 in FundMe contract.....fund function dosen't work on Remix VM....why??

### DISCUSSION THREAD

**Discussion by Raghav-Programmer** [Cyfrin/foundry-full-course-cu#359] [Q&A] [answered]
why I deployed the below code on remix vm, it deployed fine but when I added value and clicked `fund` function it gave error and reverted back. Is it because fund function contains `getConversionRate()` function which call aggregatorV3interface on sepolia eth to usd price feed??

here is fundme.sol:

` // SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {PriceConverter} from "./PriceConverter.sol";

contract FundMe {
    using PriceConverter for uint256;

    uint256 constant MIN_USD = 5e18;
    address[] funders;
    mapping(address funder => uint256 amountFunded) addressToAmountFunded; 

    address immutable i_owner;
    constructor() {
        i_owner = msg.sender;
    }

    function fund() public payable {
        require(msg.value.getConversionRate() >= MIN_USD, "Didn't Send Enough!!!");
        funders.push(msg.sender);
        addressToAmountFunded[msg.sender] += msg.value; 
    }

    function withdraw() public onlyOwner {
        for(uint256 funderIndex = 0; funderIndex < funders.length; funderIndex++) {
            addressToAmountFunded[funders[funderIndex]] = 0;
        }
        funders = new address[](0);

        //transfer
        //payable(msg.sender).transfer(address(this).balance);
        //send
        //bool sent = payable(msg.sender).send(address(this).balance);
        //require(sent, "Send Failed");
        //call
        (bool callSuccess,) = payable(msg.sender).call{value: address(this).balance}("");
        require(callSuccess, "Call Failed");
    }

    modifier onlyOwner() {
        require(msg.sender == i_owner, "Sender is Not Owner!!");
        _;
    }

    receive() external payable {
        fund();
    }

    fallback() external payable {
        fund();
    }

}`

here is priceConvertor.sol:

`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

library PriceConverter {
    
    function getPrice() public view returns(uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        (,int256 price,,,) = priceFeed.latestRoundData();
        return uint256(price*1e10);
    }

    function getConversionRate(uint256 ethAmount) internal view returns(uint256) {
        uint256 ethPrice = getPrice();
        uint256 ethInUSD = (ethAmount * ethPrice) / 1e18;
        return ethInUSD;
    }

} `

**Comment by Naman1729** [comment, marked as answer]
Hey @Raghav-Programmer , **Price Feed Contract Addresses** are only available for **Mainnet** and some of **Testnet** . but, we have no **Price Feed Contract Addresses** available for VM of remix to put into this line :

```solidity
AggregatorV3Interface priceFeed = AggregatorV3Interface(/*Price Feed Contract Addresses*/);
```

So, we use those network which have **Price Feed Contract Addresses** and you can go to this **Link** to check the **Price Feed Contract Addresses** of your desirable network are available or not.

https://docs.chain.link/data-feeds/price-feeds/addresses

**Reply by Raghav-Programmer** [reply]
Got it thanks!!!
